packages/core/src/utils/useControlled.ts

```diff
@@ -31,15 +31,15 @@ export interface UseControlledProps<T = unknown> {
  * Copied from MUI (v5) useControlled hook with one additional returned value
  * @see https://github.com/mui-org/material-ui/blob/0979e6a54ba47c278d1f535953c0520a86349811/packages/material-ui-utils/src/useControlled.js
  */
-export function useControlled<S = unknown>({
+export function useControlled<S>({
   controlled,
   default: defaultProp,
   name,
   state = "value",
 }: UseControlledProps<S>): [S, Dispatch<SetStateAction<S>>, boolean] {
   const { current: isControlled } = useRef(controlled !== undefined);
-  const [valueState, setValue] = useState(defaultProp);
-  const value = isControlled ? controlled : valueState;
+  const [valueState, setValue] = useState<S>(defaultProp);
+  const value = controlled !== undefined ? controlled : valueState;
```

Comment:
wouldn't this work? 
`const value = controlled || valueState`

Reply:
I think the more explicit way of doing this is cleaner here. I would have liked to re-use isControlled but typescript wasn't narrowing the tyeps.